crates/ruff_dev/src/format_dev.rs

```diff
@@ -209,99 +208,60 @@ pub(crate) fn main(args: &Args) -> anyhow::Result<ExitCode> {
     }
 }
 
-/// Each `path` is one of the `files` in `Args`
-enum Message {
-    Start {
-        path: PathBuf,
-    },
-    Failed {
-        path: PathBuf,
-        error: anyhow::Error,
-    },
-    Finished {
-        path: PathBuf,
-        result: CheckRepoResult,
-    },
-}
-
 /// Checks a directory of projects
 fn format_dev_multi_project(args: &Args) -> bool {
     let mut total_errors = 0;
     let mut total_files = 0;
-    let mut num_projects: usize = 0;
     let start = Instant::now();
 
-    rayon::scope(|scope| {
-        let (sender, receiver) = channel();
-
-        // Workers, to check is subdirectory in parallel
-        for base_dir in &args.files {
-            for dir in base_dir.read_dir().unwrap() {
-                num_projects += 1;
-                let path = dir.unwrap().path().clone();
+    let project_paths: Vec<_> = args
+        .files
+        .iter()
+        .flat_map(|dir| dir.read_dir().unwrap())
```

Comment:
i wouldn't unwrap those due to some failures in the past, can we collect them into an error?